.github/workflows/README.md

```diff
@@ -0,0 +1,115 @@
+# Workflow Taxonomy
+
+## Where and what tests are run
+
+---
+
+### Workflow: `deploy-happy-stack.yml`
+
+#### Ran on:
+
+- `deployment`: (deployments are triggered by `push-tests`)
+
+#### Jobs:
+
+- ##### `upgrade`: TODO
+- ##### `e2e-test`:
+  - Runs playwright against deployed frontend app uploading any artifacts
+    - ```bash
+            if [ "${DEPLOYMENT_STAGE}" == "stage" ]; then
+            export DEPLOYMENT_STAGE=staging
+          fi
+          if [ "${DEPLOYMENT_STAGE}" != "prod" ]; then
+            pip3 install -r scripts/smoke_tests/requirements.txt
+            python3 -m scripts.smoke_tests.setup
+          fi
+          cd frontend
+          npm ci
+          npx playwright install --with-deps
+          cp src/configs/${DEPLOYMENT_STAGE}.js src/configs/configs.js
+          DEBUG=pw:api npm run e2e-${DEPLOYMENT_STAGE}
+      ```
+- ##### `functional-test`:
+  - If environment is not `prod` start up backend container and run `make local-functional-test`
+  - `make local-functional-test` -> `python3 -m unittest discover --start-directory tests/functional --top-level-directory . --verbose`
+- ##### `performance-test`:
+  - If environment is `prod` run `make prod-performance-test`
+  - `make prod-performance-test` -> `python3 -m unittest discover --start-directory tests/performance --top-level-directory . --verbose`
+
+### Workflow: `lint-pr.yml`
+
+#### Ran on:
+
+- `pull_request_target`
+
+  - opening, editing, and syncing a pull request
+
+  #### Summary: Runs a GitHub action that lints the PR commit message
+
+### Workflow: `push-processing-base.yml`
+
+TODO
+
+### Workflow: `push-rdev.yml`
+
+TODO
+
+### Workflow: `push-tests.yml`
+
+#### Ran on:
+
+- `push` to `main`, `staging`, `prod`
+- `pull_request` targeting `"*"`
+
+#### Jobs:
+
+- ##### `lint`:
+
+  - Inside backend container: `make lint` -> `flake8 backend tests`
+  - Inside frontend container: (frontend/Makefile)`make lint` -> `npm run prettier-check & npm run lint`
+    - `npm run prettier-check` -> `prettier --check .`
+    - `npm run lint` -> `concurrently \"node_modules/.bin/next lint\" \"node_modules/.bin/stylelint --fix '**/*.{js,ts,tsx,css}'\" \"npm run type-check\"`
+      - `node_modules/.bin/next lint` -> runs eslint with config [next.js docs](https://nextjs.org/docs/basic-features/eslint)
+      - `node_modules/.bin/stylelint --fix '**/*.{js,ts,tsx,css}'` -> runs stylelint with config
+      - `npm run type-check` -> `tsc --noEmit`
+
+- ##### `e2e-test`:
+
+  - Installs dependencies and runs frontend server pointing at dev deployment:
```

Comment:
nit: since we have FE and BE servers, maybe the following will be more explicit?

```suggestion
  - Installs dependencies and runs frontend server pointing at dev env BE API:
```